subworkflows/local/03_assemble/assemble_organelles.nf

```diff
@@ -1,40 +1,50 @@
+include { ENTREZDIRECT_ESEARCH       } from "../../../modules/nf-core/entrezdirect/esearch/main"
 include { MITOHIFI_FINDMITOREFERENCE } from "../../../modules/nf-core/mitohifi/findmitoreference/main"
 include { MITOHIFI_MITOHIFI          } from "../../../modules/nf-core/mitohifi/mitohifi/main"
 
 workflow ASSEMBLE_ORGANELLES {
     take:
-    ch_raw_assemblies
+    ch_assembly_data // Channel: tuple(meta, assembly / reads )
+    ch_assembly_mode // String: enum('c','r')
 
     main:
     ch_versions = Channel.empty()
-    // Find mitochondria
+    // Check species name is found in NCBI taxonomy database, exit if 0 or >1 found
+    ENTREZDIRECT_ESEARCH( ch_assembly_data.map { meta, _assembly_data -> [ meta, meta.sample.name ] }.unique(), 'taxonomy' )
+
+    // Fetch mitochondrial reference sequence
     // TODO: Need to check options to mitohifi modules.
-    MITOHIFI_FINDMITOREFERENCE( ch_raw_assemblies.map { meta, _assemblies -> [ meta, meta.sample.name ] }.unique() )
-    mitohifi_ch = ch_raw_assemblies
+    MITOHIFI_FINDMITOREFERENCE( ch_assembly_data.map { meta, _assembly_data -> [ meta, meta.sample.name ] }.unique(), ENTREZDIRECT_ESEARCH.out.success )
```

Comment:
I'm not understanding something. Why are we only looking at the `success`, and not incorporating an output in the channel somehow?